src/kernel/vmm.zig

```diff
@@ -369,6 +385,82 @@ pub fn VirtualMemoryManager(comptime Payload: type) type {
             return null;
         }
 
+        ///
+        /// Copy data from an address in a virtual memory manager to an address in another virtual memory manager
+        ///
+        /// Arguments:
+        ///     IN self: *Self - The virtual memory that owns the data being copied. This must be the VMM currently in use
+        ///     IN to: *const Self - The virtual memory manager that owns the address that the data is being copied to
+        ///     IN data: []const u8 - The data being copied. Must be within memory mapped in `self`
+        ///     IN dest: usize - The address within `to` to copy the data to. The space covered by `dest` and `dest` + `dest.len` must be mapped in `to`
+        ///
+        /// Error: VmmError || pmm.PmmError || Allocator.Error
+        ///     VmmError.NotAllocated - Some or all of the destination isn't mapped
+        ///     VmmError.OutOfMemory - There wasn't enough space in the VMM to use for temporary mapping
+        ///     Bitmap(u32).Error.OutOfBounds - The address given is outside of the memory managed
+        ///     Allocator.Error.OutOfMemory - There wasn't enough memory available to fulfill the request
+        ///
+        pub fn copyDataToVMM(self: *Self, to: *const Self, data: []const u8, dest: usize) (bitmap.Bitmap(usize).BitmapError || VmmError || Allocator.Error)!void {
```

Comment:
Could this be unit tested

Reply:
We won't be able to test the functionality itself, only the function's clean-up and such. I'll give it a go.

Reply:
Sure, Also if possible try to reach all the Error.